reconcile/openshift_namespace_labels.py

```diff
@@ -0,0 +1,419 @@
+    return OC_Map(namespaces=namespaces, integration=QONTRACT_INTEGRATION,
+                  settings=settings, internal=internal,
+                  use_jump_host=use_jump_host,
+                  thread_pool_size=thread_pool_size,
+                  init_projects=True)
+
+
+class LabelError(Exception):
+    pass
+
+
+def validate_labels(labels) -> Iterable[str]:
+    """
+    Validate a label key/value against some rules from
+    https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+    """
+    if labels is None:
+        return []
+
+    err = []
+    for k, v in labels.items():
+        if len(v) > 63:
+            err.append(f'Label value longer than 63 chars: {v}')
+        pattern = '^[A-Za-z0-9][A-Za-z0-9-_\\.]{0,61}[A-Za-z0-9]$'
+        if not re.match(pattern, v):
+            err.append(f'Label value is invalid: {v}')
+
+        prefix, name = '', k
+        if '/' in k:
+            split = k.split('/')
+            if len(split) > 3:
+                err.append(f'Only one "/" allowed in label keys: {k}')
+            prefix, name = split[0], split[1]
+        if len(name) > 63:
+            err.append(f'Label key name is longer than 63 chars: {name}')
+        if not re.match(pattern, name):
+            err.append(f'Label key name is invalid: {name}')
+        if prefix:
+            if len(prefix) > 253:
+                err.append(f'Label key prefix longer than 253 chars: {prefix}')
+            hostname = ('^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*'
+                        '([A-Za-z]|[A-Za-z][A-Za-z0-9\\-]*[A-Za-z0-9])$')
+            if not re.match(hostname, prefix):
+                err.append(f'Label key prefix is invalid: {prefix}')
+            if prefix in ('kubernetes.io', 'k8s.io'):
+                err.append(f'Label key prefix is reserved: {prefix}')
+
+    return err
+
+
+def get_desired(inventory: LabelInventory, oc_map: OC_Map, namespaces) -> None:
+    """
+    Fill the provided label inventory with every desired info from the
+    input namespaces. Ocm_map is used to not register clusters which are
+    unreachable or not configured (due to --internal / --external)
+    """
+    for ns in namespaces:
+        if 'labels' not in ns:
+            continue
+        cluster, ns_name = get_names_for_namespace(ns)
+        # Skip unreachable / non-hanlded clusters
+        # eg: internal settings may not match --internal / --external param
+        if cluster not in oc_map.clusters():
+            continue
+        labels = json.loads(ns['labels'])
+        validation_errors = validate_labels(labels)
+        for err in validation_errors:
+            inventory.add_error(cluster, ns_name, err)
+        if inventory.get(cluster, ns_name, DESIRED) is not None:
+            inventory.add_error(cluster, ns_name,
+                                'Found several namespaces for ' +
+                                f'{cluster}/{ns_name}')
```

Comment:
From: https://coreos.slack.com/archives/GGC2A0MS8/p1631028931078100

To avoid cluttering the output you could just see if you have any and log an info message saying that we have reached this case and to run the integration in debug mode.

Until we can enforce uniqueness in app-interface namespace objects, we cannot do better, I won't fail the integration for this, it is not its job.